source/apiVolontaria/location/tests/tests_view_Addresses.py

```diff
@@ -0,0 +1,119 @@
+import json
+
+from django.utils import timezone
+from django.urls import reverse
+
+from rest_framework import status
+from rest_framework.test import APIClient, APITestCase
+
+from apiVolontaria.factories import UserFactory, AdminFactory
+from ..models import Address, Country, StateProvince
+
+
+class AddressesTests(APITestCase):
+
+    def setUp(self):
+        self.client = APIClient()
+
+        self.user = UserFactory()
+        self.user.set_password('Test123!')
+        self.user.save()
+
+        self.admin = AdminFactory()
+        self.admin.set_password('Test123!')
+        self.admin.save()
+
+        self.random_country = Country.objects.create(
+            name="random country",
+            iso_code="RC",
+        )
+        self.random_state_province = StateProvince.objects.create(
+            name="random state",
+            iso_code="RS",
+            country=self.random_country,
+        )
+        self.address = Address.objects.create(
+            address_line1='random address 1',
+            postal_code='RAN DOM',
+            city='random city',
+            state_province=self.random_state_province,
+            country=self.random_country,
+        )
+
+    def test_create_new_address_with_permission(self):
+        """
+        Ensure we can create a new address if we have the permission.
+        """
+        data = dict(
+            address_line1='random address 2',
+            postal_code='RAN DOM',
+            city='random city',
+            state_province=self.random_state_province.iso_code,
+            country=self.random_country.iso_code,
+        )
+
+        self.client.force_authenticate(user=self.admin)
+
+        response = self.client.post(
+            reverse('location:addresses'),
+            data,
+            format='json',
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
```

Comment:
Add an other `assertEqual` to check if we return the content created (always used by frontends to get the formatted value by the server)